web/server/codechecker_server/api/report_server.py

```diff
@@ -503,6 +509,57 @@ def get_source_component_file_query(
     return None
 
 
+def get_reports_by_bugpath_filter_for_single_origin(
+    session,
+    file_filter_q
+) -> Set[int]:
+    """
+    This function returns a query for report IDs that are related to a single
+    file described by the query in the second parameter."""
```

Comment:
file -> component?

or better:

This function returns a query for report IDs that are fully contained within the files specified by the file_filter_q query.